### ISSUE TITLE
http://xapi.jisc.ac.uk/extensions/vle_mod_id

### ISSUE THREAD

**Issue by michaelwebbjisc** [jiscdev/xapi-vle#40]
Oh dear - I end up being the first person with a patch suggestion...

Working on the moodle module, I think we need to extend context 'courseArea'.  At the moment it's got the internal id as the id, and a few people have said it would be better to have the id that they institution see.  In moodle, this in 'short name' and typical has a module code in, or something the institution can understand.  I've called it 'VLE_MOD_ID' as that's the field it links to in the UDD.

`  "http://xapi.jisc.ac.uk/extensions/courseArea": {
                "http://xapi.jisc.ac.uk/extensions/vle_mod_id": "LA101",
                "id": "http://localhost:8080/moodle/moodle/course/view.php?id=2"
            }`

**Comment by arc12** [comment]
Congratulations, you get to wear the silly hat for the rest of the day!

Can I just try to clarify the issue, and the solution with some assertions which can be agreed to, disputed etc...

1. There is an id for the course area which is (broadly) immutable in the VLE (assuming no rebuilds etc) which may be opaque but is reliable for fairly long periods of time and is likely to be the one we have access to for the purpose of matching to UDD modules. We don't mind that this is opaque if that is the price to pay for approximate reliability.

2. There is the URI which appears in xAPI statements and which refers to the same entity as assertion #1. There will be a well defined (and we hope simple) syntactic relationship to #1. This may be a direct equivalence.

3. There may be another identifier which is used by humans and which is exposed via the VLE UI. This may be a short name that is associated with #1 but is changeable according to the administrative whim of carbon-based life-forms. It is less reliable over longer terms so we won't want to use it for data processing (including mapping to UDD entity instances) unless we really have to but we might wish to resolve unfriendly identifiers to it for human consumption.

@michaelwebbjisc could you clarify the intent of your patch suggestion in relation to these (or amended) assertions. I'm particularly keen that we don't throw a spanner in the id mapping process. I'm not opposed to the requirement, but want to be clear on the solution. Our experience of the mapping for Exeter was that this is somewhat messy and the implication that a short name in Moodle-land can be so directly related to a UDD identifer makes me a little queasy.

Sorry i f I've mis-understood.

Adam

**Comment by michaelwebbjisc** [comment]
Yes, there's no perfect answer, but I think we lost this one somewhere along the way. It was actually originally a request from Chris I think, as Exeter provide a table which linked the shortname to modules, but had no understanding of the internal moodle id, which is the only item we've got in the current recipes.

Institutions can only match the internal Moodle id to something they understand if they have access to the database.   Exeter didn't as they were hosted.  We managed to get the table in the end as a one-off from ULCC,  but we can't rely on it (we haven't been able to get it for our next group).  

So if we only have the internal moodle id we may have no way of linking it to modules.

I agree with your analysis, which is why I added it as an extra item.  With both of those values present we'll definitely be able to link to modules. It also gives options for institutions who rebuild every year.
